### DISCUSSION TITLE
(ambitious?) feature request: enhancing search

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#895] [Ideas]
This feature request seems really ambitious to me. I suggest it, without having any real hope that it can be done. But if you don't ask... 

When opening the full Find and Replace dialog, for either a single file or for all open files, it would be nice if I could use regex to find not just instances within a single node, but instances within a node and all of its children.

For example, if I want to find instances of _mammal_ and _cat_, I would use mammal.*cat. But this is currently limited to a single node, where both _mammal_ and _cat_ appear in the same nodes. But I can easily see where I'd have _mammal_ in a parent node and _cat_ in one or more child nodes.  And I could see instances where _mammal_ might not be the parent, and _cat_ would still be a child, for example a parent node called _pets_.

So in my enhanced Find example, _mammal.*cat_ would find those instances of _cat_ where _mammal_ was the parent, but exclude those where _pet_ was the parent.

I hope my description is clear. If not feel free to ask questions.

And if there is already a way to do this, please let me know. That would be super helpful.

~ Bal

**Comment by euu2021** [comment]
Try the Jumper Add-on. It has a transversal search option:
![image](https://user-images.githubusercontent.com/77707706/208508847-5d93df0e-5a48-45fb-85b0-3dfd76367b9b.png)

Maybe, playing with regular expressions it is possible to also capture cat 3 in the example above.

**Reply by wordmuse** [reply]
where can I get the jumper add-on?

**Reply by euu2021** [reply]
[Jumper Add-on](https://github.com/lilive/Freeplane-Jumper) on GitHub.

**Comment by macmarrum** [comment]
I would use Filter to limit the view only to text contains `cat` where ancestor's text contains `mammal`. Only then I'd do Find and replace.

**Reply by wordmuse** [reply]
Nice - thanks for the work-around.
